### ISSUE TITLE
Wrong escaping of string literals in doubly nested EXECUTE IMMEDIATE blocks for DDL emulations

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13226]
When DDL emulations nest more than once using `EXECUTE IMMEDIATE` blocks, e.g. in Db2, then the escaping of string literals isn't correct. E.g. this recently added test for `CREATE TABLE AS SELECT` with inline index specifications:

```java
Field<Integer> c = field(name("c"), INTEGER);

create().createTable(t).column(c).index(Internal.createIndex(name("i"), t, new OrderField[] { c }, false)).as(
    select(val("v").as(c))
).execute();
```

Generates this SQL in Db2:

```sql
begin
  execute immediate '
    begin
      execute immediate ''
        create table "t" (
          "c"
        )
        as (
          select ''v'' "c" -- wrong
          from SYSIBM.DUAL
        ) with no data
      '';
      execute immediate ''
        insert into "t"
        select ''v'' "c" -- wrong again
        from SYSIBM.DUAL
      '';
    end
  ';
  execute immediate '
    create index "i" on "t"("c")
  ';
end
```

The `''v''` string literals aren't correct, they should have more escaping, namely:

```sql
begin
  execute immediate '
    begin
      execute immediate ''
        create table "t" (
          "c"
        )
        as (
          select ''''v'''' "c" -- Now fixed
          from SYSIBM.DUAL
        ) with no data
      '';
      execute immediate ''
        insert into "t"
        select ''''v'''' "c" -- Now fixed
        from SYSIBM.DUAL
      '';
    end
  ';
  execute immediate '
    create index "i" on "t"("c")
  ';
end
```

**Comment by lukaseder** [comment]
The reason is we're only escaping `'` inside of strings, not if the `'\''` character is rendered by some querypart

**Comment by lukaseder** [comment]
Possible fix:

```patch
diff --git a/jOOQ/src/main/java/org/jooq/impl/DefaultRenderContext.java b/jOOQ/src/main/java/org/jooq/impl/DefaultRenderContext.java
index a0a2a0f..bd19913 100644
--- a/jOOQ/src/main/java/org/jooq/impl/DefaultRenderContext.java
+++ b/jOOQ/src/main/java/org/jooq/impl/DefaultRenderContext.java
@@ -454,9 +454,10 @@
     @Override
     public final RenderContext sql(char c) {
         applyNewLine();
-        sql.append(c);
 
         if (c == '\'' && stringLiteral())
+            sql.append(stringLiteralEscapedApos);
+        else
             sql.append(c);
 
         resetSeparatorFlags();
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13228), 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13229)
